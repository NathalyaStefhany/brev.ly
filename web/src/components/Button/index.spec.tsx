import { describe, expect, it } from 'vitest';
import { render, screen } from '@testing-library/react';
import { Trash } from '@phosphor-icons/react';

import { Button } from '.';

describe('Button tests', () => {
  it('should render correctly', () => {
    expect(render(<Button>Label</Button>));
  });

  it('should display the label text correctly', () => {
    render(<Button data-testid="button">Label</Button>);

    expect(screen.getByTestId('button')).toBeInTheDocument();
    expect(screen.getByTestId('button')).toHaveTextContent('Label');
  });

  it('should display the icon when provided', () => {
    render(
      <Button icon={Trash} data-testid="button">
        Label
      </Button>,
    );

    expect(screen.getByTestId('button-icon')).toBeInTheDocument();
  });

  it('should not display the icon', () => {
    render(<Button data-testid="button">Label</Button>);

    expect(screen.queryByTestId('button-icon')).not.toBeInTheDocument();
  });

  it('should not display the icon when is loading', () => {
    render(
      <Button icon={Trash} isLoading data-testid="button">
        Label
      </Button>,
    );

    expect(screen.queryByTestId('button-icon')).not.toBeInTheDocument();
    expect(screen.getByTestId('button-loading')).toBeInTheDocument();
  });

  it('should display primary button correctly', () => {
    const { container } = render(
      <Button variant="primary" data-testid="button">
        Label
      </Button>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should display secondary button correctly', () => {
    const { container } = render(
      <Button variant="secondary" data-testid="button">
        Label
      </Button>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should display loading button correctly', () => {
    const { container } = render(
      <Button isLoading data-testid="button">
        Label
      </Button>,
    );

    expect(container).toMatchSnapshot();
  });
});
