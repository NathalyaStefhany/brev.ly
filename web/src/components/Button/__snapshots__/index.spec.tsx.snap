// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button tests > should display loading button correctly 1`] = `
<div>
  <button
    class="flex flex-row gap-3 items-center justify-center cursor-pointer transition-all duration-300 disabled:cursor-not-allowed max-w-176 w-full bg-blue-base text-md text-white rounded-lg px-10 py-7.5 hover:not-disabled:bg-blue-dark disabled:opacity-50"
    data-testid="button"
    type="button"
  >
    <div
      class="w-6 h-6 border-2 border-gray-100 border-t-white/40 rounded-full animate-spin"
      data-testid="button-loading"
    />
    Label
  </button>
</div>
`;

exports[`Button tests > should display primary button correctly 1`] = `
<div>
  <button
    class="flex flex-row gap-3 items-center justify-center cursor-pointer transition-all duration-300 disabled:cursor-not-allowed max-w-176 w-full bg-blue-base text-md text-white rounded-lg px-10 py-7.5 hover:not-disabled:bg-blue-dark disabled:opacity-50"
    data-testid="button"
    type="button"
  >
    Label
  </button>
</div>
`;

exports[`Button tests > should display secondary button correctly 1`] = `
<div>
  <button
    class="flex flex-row gap-3 items-center justify-center cursor-pointer transition-all duration-300 disabled:cursor-not-allowed bg-gray-200 border border-gray-200 text-sm font-semibold text-gray-500 rounded-sm p-3.5 hover:not-disabled:border-blue-base disabled:opacity-50"
    data-testid="button"
    type="button"
  >
    Label
  </button>
</div>
`;
